agent/consul/stream/event_publisher_test.go

```diff
@@ -233,7 +237,22 @@ func TestEventPublisher_SubscribeWithIndexNotZero_CanResume(t *testing.T) {
 
 	publisher := NewEventPublisher(newTestSnapshotHandlers(), time.Second)
 	go publisher.Run(ctx)
-	// Include the same event in the topicBuffer
+
+	// Start a separate subscription that remains open throughout the entire test
+	// to prevent the buffer getting cleaned up because there aren't any subscribers.
+	sub, err := publisher.Subscribe(req)
+	require.NoError(t, err)
+	defer sub.Unsubscribe()
+
+	// Publish the testSnapshotEvent, to ensure that it is skipped over when
+	// splicing the topic buffer onto the snapshot.
+	//
+	// Note: we must evict the snapshot that was created by the above subscription
+	// first, as it has already had the topic buffer spliced onto it (so therefore
+	// would include the snapshot event again).
+	publisher.lock.Lock()
+	delete(publisher.snapCache, req.topicSubject())
+	publisher.lock.Unlock()
```

Comment:
In normal operation the cache TTL would perform this operation, is that right? So the scenario this is setting up is when the cache has expired, but we still have an active subscribe?  

If that is correct, I wonder if this would be better as a helper function we can call from all 3 tests. The godoc comment on that function would be a great place to document that we are doing this to setup a specific scenario ("a cache has exceeded the TTL, but there are still active subscriptions"). We could use `t.Cleanup` to call the `sub.Unsubscribe` instead of the defer.

Reply:
Great point! I've swapped this out for a test helper called `simulateExistingSubscriber` which I think captures the intent better (it actually took me a while to remember why I did this 😅).